client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -96,7 +96,10 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 	);
 
 	const { data: generatedDesigns = [], isLoading: isLoadingGeneratedDesigns } =
-		useStarterDesignsGeneratedQuery( { seed: siteSlug || undefined } );
+		useStarterDesignsGeneratedQuery( {
+			vertical_id: siteVerticalId,
```

Comment:
Unfortunately, `siteVerticalId` will be `undefined` while the component fetches the value from API, so `vertical_id` never gets sent. I think we can add the vertical ID to the `useStarterDesignsGeneratedQuery`'s query key to force it to update when `siteVerticalId` changes.

Reply:
How about just setting the vertical title from the client side? I think it would be better if we have the site title step before the generated design picker and want to show it in the preview. What do you think?

Reply:
@arthur791004: Even if we set the vertical title from Calypso (which needs some works as vertical title isn't currently stored in site settings), it will need to pass `vertical_id` anyway for other use cases. E.g. determining if the backend can verticalize the images. So this logic will still need to be shipped.

I'd prefer do your suggestion (passing custom site title from client side) later when we actually implement custom site title user input step.

What do you think?

Reply:
Yes, for context, @fushar is correct in that we would need an additional API call to retrieve the vertical title since it's currently not stored in site settings.

Reply:
Okay, that's fine.

Reply:
> I think we can add the vertical ID to the useStarterDesignsGeneratedQuery's query key to force it to update when siteVerticalId changes.

@taipeicoder Nice catch! I've added the whole query params to the cache key, and retested it. I think it works. Could you test again?

Reply:
Work! This is a minor nitpick, but do you (both) think we should set `enabled` only when `vertical_id` is present to avoid an extra API call? In theory, the site should have `vertical_id` set in order to be eligible for the v13n DP.

Reply:
Ah, I see. Makes sense. But can this logic be added to @arthur791004's PR instead? I think it is more relevant to be added there, as this PR is just for getting vertical title in the already existing endpoint call 🙈 😂

Reply:
Sounds good to me 😆

Reply:
my PR?

Reply:
Sorry, I mean https://github.com/Automattic/wp-calypso/pull/64140

Reply:
As this PR added the `vertical_id` to the hook and change the current behavior, I think it should also update the enabled behavior IMO🙈

Reply:
Oh that also makes sense to me to add in that PR as I also need to add the vertical id to this hook 🤣